packages/nx/src/native/workspace/get_config_files.rs

```diff
@@ -1,17 +1,100 @@
 use crate::native::parallel_walker::nx_walker;
 use crate::native::utils::glob::build_glob_set;
+use crate::native::utils::path::Normalize;
+use globset::GlobSet;
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
 
 #[napi]
 /// Get workspace config files based on provided globs
 pub fn get_config_files(workspace_root: String, globs: Vec<String>) -> anyhow::Result<Vec<String>> {
     let globs = build_glob_set(globs)?;
     Ok(nx_walker(workspace_root, move |rec| {
-        let mut config_paths: Vec<String> = vec![];
-        for (path, _) in rec {
-            if globs.is_match(&path) {
-                config_paths.push(path.to_owned());
-            }
+        let mut config_paths: HashMap<PathBuf, (PathBuf, Vec<u8>)> = HashMap::new();
+        for (path, content) in rec {
+            insert_config_file_into_map((path, content), &mut config_paths, &globs);
         }
         config_paths
+            .into_iter()
+            .map(|(_, (val, _))| val.to_normalized_string())
+            .collect()
     }))
 }
+
+pub fn insert_config_file_into_map(
+    (path, content): (PathBuf, Vec<u8>),
+    config_paths: &mut HashMap<PathBuf, (PathBuf, Vec<u8>)>,
+    globs: &GlobSet,
+) {
+    if globs.is_match(&path) {
+        let parent = path.parent().unwrap_or(Path::new("")).to_path_buf();
```

Comment:
use `unwrap_or_else` so that it's lazily evaluated. 
```suggestion
        let parent = path.parent().unwrap_or_else(|| Path::new("")).to_path_buf();
```